src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,125 @@
+                    book = mapToBook(resultSet);
+                }
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find by id " + id, e);
+        }
+        return Optional.ofNullable(book);
+    }
+
+    @Override
+    public List<Book> findAll() {        
+        List<Book> bookList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(SELECT_ALL_QUERY);
+                                ResultSet resultSet = preparedStatement.executeQuery()) {
+            while (resultSet.next()) {
+                bookList.add(mapToBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find any row", e);
+        }
+        return bookList;
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(UPDATE_QUERY, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setInt(2, book.getPrice().intValue());
+            preparedStatement.setLong(3, book.getId());
+            checkAffectedRows(preparedStatement.executeUpdate());
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(DELETE_QUERY, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete by id " + id, e);
+        }
+    }
+
+    private Book mapToBook(ResultSet resultSet) throws SQLException {
+        return new Book(
+                resultSet.getLong("id"),
+                resultSet.getString("title"),
+                resultSet.getBigDecimal("price"));
+    }
+
+    private void checkAffectedRows(int affectedRows) {
+        if (affectedRows < 1) {
+            throw new DataProcessingException("Expected to insert at least 1 row "
+                    + "but inserted 0 rows");
+        }
+    }
+
+    private static void setCreatedBookId(Book book, PreparedStatement preparedStatement)
+            throws SQLException {
+        try (ResultSet generatedKeys = preparedStatement.getGeneratedKeys()) {
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getLong("id");
+                book.setId(id);
+            }
+        }
+    }

```

Comment:
1. this method shouldn't be static
2. passing some object into a method and changing it somehow is a bad practice. this method should instead return the generated key and you can use that return value as you wish